src/adc.rs

```diff
@@ -136,7 +137,9 @@ impl Temperature {
 macro_rules! adc_pins {
     ($($pin:ty => ($adc:ident, $chan:expr)),+ $(,)*) => {
         $(
-            impl Channel<stm32::$adc> for $pin {
+            impl<P> Channel<stm32::$adc> for P
+                where P: stasis::EntitlementLock<Resource = $pin>
+            {
```

Comment:
This does not work. Both `Channel` and `EntitlementLock` are from outside this crate. There are also errors about conflicting implementations with the other opamps below.

Reply:
I think the simplest solution is to make a local trait like `AdcChannel` which requires `embedded_hal::adc::Channel` and then it would work.

I'm not very familiar with what you're doing here though. I'll think on it.

You could also do concrete implementations since this is macro generated anyway.

Reply:
> You could also do concrete implementations since this is macro generated anyway.

Yeah that would probably be the simplest solution. I'll give it a go